src/rust/lib/graph-editor/src/component/cursor.rs

```diff
@@ -56,23 +129,34 @@ pub mod shape {
     use super::*;
 
     ensogl::define_shape_system! {
-        (position:Vector2<f32>, width:f32, height:f32, selection_size:Vector2<f32>, press:f32, radius:f32) {
+        ( position       : Vector2<f32>
+        , dim            : V2
+        , offset         : V2
+        , selection_size : Vector2<f32>
+        , press          : f32
+        , radius         : f32
+        , color          : Vector4<f32>
+        ) {
```

Comment:
Those are like parameters of function? If yes, the linter probably will compain about too-many-function-arguments, and obligate you to decompose the arguments to structs. And I agree with that.

Reply:
These are parameters to glsl function. You cannot decompose them here :)